File: 1370-count-number-of-nice-subarrays/count-number-of-nice-subarrays.py
class Solution:
    def numberOfSubarrays(self, nums: List[int], k: int) -> int:
        n = len(nums)
        count = [0]*(n+1)
        count[0] = 1
        ans = 0
        t = 0
        for i in nums:
            t += i&1
            if t-k >= 0:
                ans += count[t-k]
            count[t] += 1
        return ans

File: 1370-count-number-of-nice-subarrays/count-number-of-nice-subarrays.cpp
class Solution {
public:
    int numberOfSubarrays(vector<int>& nums, int k) {
        int n = nums.size();
        vector<int>total_count(n+1,0);
        total_count[0] = 1;
        int count = 0, ans = 0;
        for(int i : nums){
            count += (i&1);
            if(count - k >= 0) ans += total_count[count-k];
            total_count[count]++;
        }
        return ans;
    }
};